src/main/java/io/confluent/connect/jdbc/dialect/OracleDatabaseDialect.java

```diff
@@ -76,18 +82,44 @@ protected String checkConnectionQuery() {
     return "SELECT 1 FROM DUAL";
   }
 
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
   @Override
   protected boolean maybeBindPrimitive(
       PreparedStatement statement,
       int index,
       Schema schema,
-      Object value
+      Object value,
+      ColumnDefinition colDef
   ) throws SQLException {
+    if (colDef == null) {
+      return super.maybeBindPrimitive(statement, index, schema, value, colDef);
+    }
+
     if (schema.type() == Type.STRING) {
-      statement.setNString(index, (String) value);
+      if (colDef.type() == Types.CLOB || colDef.type() == Types.NCLOB) {
+        statement.setCharacterStream(index, new StringReader((String) value));
+        return true;
+      } else if (colDef.type() == Types.NVARCHAR || colDef.type() == Types.NCHAR) {
+        statement.setNString(index, (String) value);
+        return true;
+      } else {
+        return super.maybeBindPrimitive(statement, index, schema, value, colDef);
+      }
+    }
+
+    if (schema.type() == Type.BYTES && colDef.type() == Types.BLOB) {
```

Comment:
Is there a reason to make BYTES data insertion into a non-BLOB column an error in the connector instead of the driver? surely the driver should know better than us what type conversions are possible.